implementations/micrometer-registry-influx/src/main/java/io/micrometer/influx/InfluxMeterRegistry.java

```diff
@@ -75,43 +85,86 @@ public static Builder builder(InfluxConfig config) {
         return new Builder(config);
     }
 
-    private void createDatabaseIfNecessary() {
-        if (!config.autoCreateDb() || databaseExists)
+    private void createStorageIfNecessary() {
+        if (!config.autoCreateDb() || storageExists)
             return;
 
+        switch (influxDBVersion) {
+            case V1:
+                createDatabaseIfNecessary();
+                break;
+            case V2:
+                createBucketIfNecessary();
+                break;
+        }
+    }
+
+    private void createDatabaseIfNecessary() {
+
         try {
             String createDatabaseQuery = new CreateDatabaseQueryBuilder(config.db()).setRetentionDuration(config.retentionDuration())
                     .setRetentionPolicyName(config.retentionPolicy())
                     .setRetentionReplicationFactor(config.retentionReplicationFactor())
                     .setRetentionShardDuration(config.retentionShardDuration()).build();
 
-            httpClient
+            HttpSender.Request.Builder requestBuilder = httpClient
                     .post(config.uri() + "/query?q=" + URLEncoder.encode(createDatabaseQuery, "UTF-8"))
-                    .withBasicAuthentication(config.userName(), config.password())
+                    .withBasicAuthentication(config.userName(), config.password());
+            influxDBVersion.addHeaderToken(this, requestBuilder);
+
+            requestBuilder
                     .send()
                     .onSuccess(response -> {
                         logger.debug("influx database {} is ready to receive metrics", config.db());
-                        databaseExists = true;
+                        storageExists = true;
                     })
                     .onError(response -> logger.error("unable to create database '{}': {}", config.db(), response.body()));
         } catch (Throwable e) {
             logger.error("unable to create database '{}'", config.db(), e);
         }
     }
 
+    private void createBucketIfNecessary() {
+
+        try {
+            String createBucketJSON = new CreateBucketJSONBuilder(config.bucket(), org)
+                    .setEverySeconds(config.retentionDuration())
+                    .build();
+
+            HttpSender.Request.Builder requestBuilder = httpClient
+                    .post(config.uri() + "/api/v2/bucket");
```

Comment:
```suggestion
                    .post(config.uri() + "/api/v2/buckets");
```

Unfortunately, it seems the Influx v2 API returns a 200 response code even though the URI is wrong. We should add some tests with an InfluxDB instance running, using testcontainers, to ensure things are working as expected now and in the future. That doesn't need to block this pull request; I can follow-up with that work after merging this.

Reply:
Fixed `bucket` -> `buckets`. Thanks.

I will prepare an integration test over https://www.testcontainers.org/modules/databases/influxdb/.

Reply:
The support for creating `Bucket` is removed. 

Do you want to create integration tests over `testcontainers` for publishing metrics?

Reply:
I've created #2504 to track adding the integration tests. I don't want to block merging this enhancement on that. We want to include support for InfluxDB 2 in the 1.7 milestone release tomorrow if possible.

Reply:
Ok